core/src/main/java/org/infinispan/commands/tx/PrepareCommand.java

```diff
@@ -230,6 +231,9 @@ public boolean isReturnValueExpected() {
       }
       Set<Object> set = new HashSet<>(modifications.length);
       for (WriteCommand wc : modifications) {
+         if (wc.hasFlag(Flag.SKIP_LOCKING)) {
```

Comment:
I don't think so. If we have 2 keys with the same hash code, one of them would not be inserted in the TreeSet because the comparator would return 0.